server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsIT.java

```diff
@@ -111,29 +173,307 @@ public void testStatsResponseFromLocalNode() {
                 .prepareRemoteStoreStats(INDEX_NAME, null);
             remoteStoreStatsRequestBuilder.setLocal(true);
             RemoteStoreStatsResponse response = remoteStoreStatsRequestBuilder.get();
-            assertTrue(response.getSuccessfulShards() == 1);
-            assertTrue(response.getShards() != null && response.getShards().length == 1);
-            RemoteRefreshSegmentTracker.Stats stats = response.getShards()[0].getStats();
-            assertResponseStats(stats);
+            assertEquals(1, response.getSuccessfulShards());
+            assertTrue(response.getRemoteStoreStats() != null && response.getRemoteStoreStats().length == 1);
+            RemoteRefreshSegmentTracker.Stats stats = response.getRemoteStoreStats()[0].getStats();
+            validateUploadStats(stats);
+        }
+        changeReplicaCount(1);
+        for (String node : nodes) {
+            RemoteStoreStatsRequestBuilder remoteStoreStatsRequestBuilder = client(node).admin()
+                .cluster()
+                .prepareRemoteStoreStats(INDEX_NAME, null);
+            remoteStoreStatsRequestBuilder.setLocal(true);
+            RemoteStoreStatsResponse response = remoteStoreStatsRequestBuilder.get();
+            assertTrue(response.getSuccessfulShards() > 0);
+            assertTrue(response.getRemoteStoreStats() != null && response.getRemoteStoreStats().length != 0);
+            for (RemoteStoreStats stat : response.getRemoteStoreStats()) {
+                ShardRouting routing = stat.getShardRouting();
+                RemoteRefreshSegmentTracker.Stats stats = stat.getStats();
+                if (routing.primary()) {
+                    validateUploadStats(stats);
+                } else {
+                    validateDownloadStats(stats);
+                }
+            }
```

Comment:
We can do a minor addition to this test - We can verify that the shards where the upload & download fields are showing as expected. Basically, assert that the node as per _cat/shards is what we see in the response as well.